semgrep_output_v1.atd

```diff
@@ -868,6 +868,124 @@ type cli_output_extra = {
     ~skipped_rules: skipped_rule list;
 }
 
+(*****************************************************************************)
+(* Matching diagnosis *)
+(*****************************************************************************)
+
+(* A "matching diagnosis" is a postprocessed interpretation of matching
+   explanations, specific to a particular test-annotated target file.
+
+   For instance, suppose we have the rule:
+   1 | all:
+   2 | - pattern: foo(...)
+   3 | - not: foo(good)
+   and the following Python annotated target:
+
+   1 | # ruleid: my_rule
+   2 | foo()
+   3 | # ok: my_rule
+   4 | foo(bad)
+
+   We would clearly get an unexpected match on line 4, which would fail
+   the test assertion.
+
+   By looking at the matching explanation, we can deduce that the match
+   on line 4 must clearly have been introduced by the positive `foo(..)`
+   pattern. The rule-writer probably meant to kill `foo(bad)` with the
+   negative `foo(good)` pattern.
+
+   This is essentially what matching diagnoses are -- using matching
+   explanations to point out where the erroneous parts of the rule _may_
+   be.
+
+   Note that this is a _may_, because an unexpected match could have been
+   killed by the `foo(bad)`, but if there were more negative patterns,
+   it could have been killed elsewhere too. All we can do is point out
+   places where the rule-writer _may_ have messed up.
+
+   So in this case, we would expect an `unexpected_match_diagnosis` with
+   the form:
+   { matched_text = { line = 4; text = "foo(bad)" };
+     originating_kind = Xpattern;
+     originating_text = { line = 2; text = "- pattern: foo(...)" };
+     killing_parents = [ { killing_parent_kind = Negation;
+                           snippet = { line = 3; text = "- not: foo(good)" } } ]
+   }
+ *)
+type matching_diagnosis = {
+  target: fpath;
```

Comment:
so this is a test target file, maybe add that in comment; it's not any target, it's a test target.